src/libraries/System.Collections.Immutable/src/System.Collections.Immutable.csproj

```diff
@@ -102,5 +119,9 @@
   </ItemGroup>
   <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
     <PackageReference Include="System.Memory"  Version="$(SystemMemoryVersion)" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Runtime.CompilerServices.Unsafe\src\System.Runtime.CompilerServices.Unsafe.ilproj" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Runtime.CompilerServices.Unsafe\src\System.Runtime.CompilerServices.Unsafe.ilproj" PrivateAssets="all" />
```

Comment:
This isn't a sustainable solution. We can't mark every P2P that points to an inbox project manually with PrivateAssets=all.

We should modify this target not just account for test projects: https://github.com/dotnet/runtime/blob/a2665c73fa479fff09a29afb58186bedee7545d8/eng/references.targets#L38-L50.

Reply:
Problem with adding PrivateAssets to all inbox P2Ps is that we would loose the transitive dependency feature which would force us to list every dependency, even if it's a transitive one, manually. I don't think we want that. Is there a way to filter out ProjectReferences without annotating them with PrivateAssets?

Reply:
Looks like the overall issue is tracked in https://github.com/NuGet/Home/issues/3891. I don't think it's possible today to not use PrivateAssets and exclude ProjectReferences/PackageReferences when using NuGet's Pack task.

I'm ok with enabling PrivateAssets meanwhile but hopefully there will be a solution provided by NuGet to suppress dependencies without loosing the transitive feature.

Reply:
pack picks this value from the assets file and any ideas how we can automize it before restore for net6.0 ?

Reply:
> any ideas how we can automize it before restore for net6.0 ?

Make sure that the target runs before the ProjectReference items are collected. This should work but might require a reaction in some of the project files:

```diff
diff --git a/eng/references.targets b/eng/references.targets
index 6adcba93322..b486a7afeb8 100644
--- a/eng/references.targets
+++ b/eng/references.targets
@@ -1,4 +1,4 @@
-<Project>
+<Project InitialTargets="UpdateProjectReferencesWithPrivateAttribute">
   <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
     <!--
       Disable RAR from transitively discovering dependencies for references. This is required as we don't copy
@@ -37,16 +37,13 @@
 
   <!-- Make shared framework assemblies not app-local (non private). -->
   <Target Name="UpdateProjectReferencesWithPrivateAttribute"
-          AfterTargets="AssignProjectConfiguration"
-          BeforeTargets="PrepareProjectReferences"
           Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
-                     ('$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true') and
-                     '@(ProjectReferenceWithConfiguration)' != ''">
+                     '$(IsReferenceAssembly)' != 'true' and
+                     '@(ProjectReference)' != ''">
     <ItemGroup>
-      <ProjectReferenceWithConfiguration PrivateAssets="all"
-                                         Private="false"
-                                         Condition="$(NetCoreAppLibrary.Contains('%(Filename);'))" />
+      <ProjectReference PrivateAssets="all"
+                        Private="false"
+                        Condition="$(NetCoreAppLibrary.Contains('%(Filename);'))" />
     </ItemGroup>
   </Target>
-
 </Project>
```

Reply:
The issue is TargetFrameworkIdentifier wont be set in the outerbuild and we wont be able to add it just for netcoreapp config.

Reply:
It worked, but i dont understand why, wont the restore happen in the outerbuild and TargetFrameworkIdentifier would be empty ?

Reply:
"Restore" collects inputs from both the outer- and the inner build. In case you are interested, here's the implementation: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs.